### PR TITLE
Refactor exercise 5

### PR DESCRIPTION
**Pull request by makcandrov**

### CODE REVIEW

**Review by Tristan22400 — APPROVED** [review]

File: src/Hack.sol [hunk 1]
```diff
@@ -112,55 +112,23 @@ contract Vault {
...
-        require(block.timestamp > timeB + 1 days);
-        require(address(this).balance >= 2 ether);
-        partyB.transfer(2 ether);
+/// @dev Contract for locking and unlocking funds using a commitment and password.
+contract Locker {
+    bytes32 commitment;
+
+    /// @dev Locks the funds by setting the commitment.
+    /// @param _commitment The commitment to lock the funds.
+    function lock(bytes32 _commitment) external payable {
```
Comment (MerlinEgalite) [inline src/Hack.sol:121]:
```suggestion
    function lock(bytes32 commitment) external payable {
```

we could move to this pattern tbh (to do in another PR)

File: src/Hack.sol [hunk 2]
```diff
@@ -112,55 +112,23 @@ contract Vault {
...
-        else partyA.transfer(2 ether);
-    }
-
-    /// @dev Time out party A if it takes more than 1 day to reveal.
-    ///      Sends ETH to party B.
-    function timeOut() public {
-        require(block.timestamp > timeB + 1 days);
-        require(address(this).balance >= 2 ether);
-        partyB.transfer(2 ether);
+/// @dev Contract for locking and unlocking funds using a commitment and password.
```
Comment (MerlinEgalite) [inline src/Hack.sol:115]: Maybe add that we suppose that there's some ETH on the contract

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

File: src/Hack.sol (hunk 1 above)
Comment (makcandrov) [reply src/Hack.sol:121]: the problem is that the storage variable is also named `commitment`

Comment (MerlinEgalite) [reply src/Hack.sol:121]: Oh yes ok I understand

**Review by MerlinEgalite — APPROVED** [review]

**Review by Tristan22400 — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]
